### PR TITLE
[#2258] Specify the content-type of a telemetry/event message via MQTT property bag

### PR DESCRIPTION
**Pull request by kaniyan**

### CODE REVIEW

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttContext.java [hunk 1]
```diff
@@ -110,6 +111,9 @@ public static MqttContext fromPublishPacket(
                     .ifPresentOrElse(propertyBag -> {
                         result.topic = ResourceIdentifier.fromString(propertyBag.topicWithoutPropertyBag());
                         result.propertyBag = propertyBag;
+                        // sets the content type using the corresponding value from the property bag
+                        Optional.ofNullable(propertyBag.getProperty(MessageHelper.SYS_PROPERTY_CONTENT_TYPE))
```
Comment (sophokles73) [inline adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttContext.java:115]: What if I specify the content type using `Content-Type`?

File: site/documentation/content/user-guide/mqtt-adapter.md [hunk 2]
```diff
@@ -97,6 +97,7 @@ This requires that
 
 The protocol adapter checks the configured [message limit] ({{< relref "/concepts/resource-limits.md" >}}) before accepting any telemetry messages. If the message limit is exceeded or the incoming telemetry message cannot be processed, the connection to the client is closed.
 
+The devices can optionally indicate the content type of the payload by setting the *content-type* property explicitly in the `property-bag`. The `property-bag` is an optional collection of properties intended for the receiver of the message. A property bag is only allowed at the very end of a topic. It always starts with a `/?` character, followed by pairs of URL encoded property names and values that are separated by `&`. For example, a property bag containing two properties *seqNo* and *importance* looks like this: `/topic/name/?seqNo=10034&importance=high`.
```
Comment (sophokles73) [inline site/documentation/content/user-guide/mqtt-adapter.md:100]: can you please insert an empty line between this paragraph and the next section heading?

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttContext.java (hunk 1 above)
Comment (kaniyan) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttContext.java:115]: Good point. Field names in [HTTP headers ](https://tools.ietf.org/html/rfc2616) are case-insensitive. I think it makes sense also to treat the property names as case-insensitive in the MQTT _property-bag_.  WDYT?

Comment (kaniyan) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttContext.java:115]: Now I have made it case insensitive and pushed the changes.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Comment by sophokles73** [comment]
@kaniyan if this is supposed to go into 1.5.0, please make sure to assign this PR and the corresponding issue to the 1.5.0 milestone and project ...

**Pull request merged**